### ISSUE TITLE
netcat-openbsd Hard to Get Installed on Centos 7

### ISSUE THREAD

**Issue by byteknacker** [saltstack-formulas/jenkins-formula#47]
First off, many thanks for putting the effort to create this formula, I am using it at my work and it is a bliss.

The problem:

When I used jenkins.plugins to install the plugins for Jenkins, jenkins_listening returns false since netcat-openbsd is not installed on Centos 7. I tried to install it but without success and it seems to be an ongoing problem that it is hard to get netcat running on Centos and Redhat distributions.

Error Message from salt '*' state.highstate:

```
ID: jenkins_listening
Function: pkg.installed
Name: netcat-openbsd
Result: False
Comment: The following package(s) were not found, and no possible matches were found in the package db: netcat-openbsd
Started: 09:21:59.906133
Duration: 14024.736 ms
```

Proposed solution:
1. Either put a standardised way of installing netcat on Centos and Redhat distributions and everything else remain the same.
2. Or to use another tool for checking the HTTP connection with jenkins server, maybe telnet or just curl

Current solution of mine:

I am using solution 2 and just use curl to test if connection with jenkins is there. I have to test if that works. I am extending your formula part for jenkins_listening.

**Comment by gravyboat** [comment]
I would be fine with switching this over to using curl, package wise this could easily be managed via a map.jinja update and a variable reference with a default of netcat-openbsd (so we do not break backwards compatibility).

**Comment by byteknacker** [comment]
That would be great, I actually used curl in my extend statement myself and it works fine.

Also I found out that there is netcat on Centos 7 and it is called nmap-netcat. 

You can follow this SO article to install it: [How to Install netcat on Centos 7](http://stackoverflow.com/questions/29045877/centos-7-and-puppet-unable-to-install-nc)

I would really prefer if it is curl though since almost everyone has it and it is better known by many developers. Btw nc -z does not work on Centos 7. The netcat cli is not as powerful on Centos 7 as on Ubuntu.

**Comment by byteknacker** [comment]
I have written this in an extend statement to overwrite the current default behaviour of jenkins_listening which is using nc. 

I am curious about your opinion as to whether just 'curl' without any options would suffice. If you recommend any option which one should I use? 

```
  jenkins_listening:
    pkg.installed:
      - name: curl
    cmd.wait:
      - name: "until curl http://localhost:{{ jenkins.jenkins_port }}; do sleep 1; done"
      - timeout: 10
      - require:
        - service: jenkins
      - watch:
        - service: jenkins
```

Today, I am working on this on my fork: https://github.com/byteknacker/jenkins-formula

I try to submit a PR today as well. Will be making it backward compatible as you suggested. So I need to change map.jinja and let users use grain to automatically get the OS?

**Comment by myoung34** [comment]
Resolved with pillar `netcat_pkg`

**Comment by sroegner** [comment]
I just ran into this when trying to install on CentOS7 - changed `netcat_pkg` to nmap-ncat as suggested and then jenkins.cli.jenkins_listening fails trying to `nc -z` which I suppose could be replaced with `nc --recv-only` on RHEL/CentOS. Am i missing something here, is there a workaround?

**Comment by myoung34** [comment]
This is resolved by #69
